src/sync_protocol/src/agent_sync_protocol.cpp

```diff
@@ -42,22 +44,21 @@ void AgentSyncProtocol::synchronizeModule(const std::string& module, Mode mode,
         return;
     }
 
-    uint64_t session = 0;
     std::vector<PersistedData> data = m_persistentQueue->fetchAll(module);
 
-    if (!sendStartAndWaitAck(module, mode, realtime, session, data.size()))
+    if (!sendStartAndWaitAck(module, mode, realtime, data.size()))
     {
         std::cerr << "Failed to send start message" << std::endl;
         return;
     }
 
-    if (!sendDataMessages(module, session, data))
+    if (!sendDataMessages(module, m_syncState.session, data))
     {
         std::cerr << "Failed to send data messages" << std::endl;
         return;
     }
 
-    if (!sendEndAndWaitAck(module, session))
+    if (!sendEndAndWaitAck(module, m_syncState.session))
     {
         std::cerr << "Failed to send end message" << std::endl;
         return;
```

Comment:
Should we call `m_syncState.reset()` at the end of this function?

Reply:
Done!